dali/operators/decoder/video/video_decoder_base.h

```diff
@@ -123,6 +123,13 @@ class DLL_PUBLIC VideoDecoderBase {
 
 
   std::vector<std::unique_ptr<FramesDecoder>> frames_decoders_;
+
+ private:
+  using storage_backend_for_copy_kernel = std::conditional_t<
+          std::is_same_v<OutBackend, CPUBackend>,
+          StorageCPU,
+          StorageGPU
+  >;
```

Comment:
There's already `backend_to_storage_device`. Would be better to use existing tools

Reply:
```
using OutStorage = backend_to_storage_device<OutBackend>;
```
^^ I think `OutStorage` is simpler.

Reply:
The problem is, that the `kernels::copy` implementation assumes using `StorageCPU`, `StorageGPU`, etc. Therefore there are the following problems:
1. Plain `OutStorage` cannot be used, because it's `...Backend` not `Storage...`.
1. `backend_to_storage_device` does not provide any information about `Storage...`. It returns `StorageDevice` enum, which is something different and not compatible with `kernels::copy` implementation.

I didn't implement any generic `StorageDevice -> Storage...` mapping, since I didn't want to introduce another feature in this PR, its big enough already.

Reply:
My bad, the one I meant is `storage_tag_map`, from views.h

Reply:
Unfortunately, `storage_tag_map` cannot be used. For the GPU operator there will not be `VideoInput<GPUBackend>`, but `VideoInput<MixedBackend>`. And the problem is, that a mapping `MixedBackend -> StorageGPU` shouldn't be introduced as a general one.